cdap-ui/app/cdap/components/AbstractWidget/SqlConditionsWidget/Rule.tsx

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License'); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import Paper from '@material-ui/core/Paper';
+import withStyles, { WithStyles, StyleRules } from '@material-ui/core/styles/withStyles';
+import IconButton from '@material-ui/core/IconButton';
+import AddIcon from '@material-ui/icons/AddBox';
+import CancelIcon from '@material-ui/icons/Cancel';
+
+import Select from 'components/AbstractWidget/SelectWidget';
+import If from 'components/If';
+
+import { IInputSchema, IRule } from 'components/AbstractWidget/SqlConditionsWidget';
+
+const styles = (theme): StyleRules => {
+  return {
+    root: {
+      display: 'flex',
+      flexAlign: 'row',
+      width: '100%',
+    },
+    rule: {
+      position: 'relative',
+      width: '100%',
+      marginBottom: '15px',
+      padding: '15px',
+    },
+    actionButtons: {
+      display: 'flex',
+      alignItems: 'flex-end',
+      marginBottom: '15px',
+      width: '50px',
+      paddingLeft: '10px',
+    },
+    stageRow: {
+      display: 'grid',
+      gridTemplateColumns: '45% 45% 10%',
+      marginBottom: '5px',
+    },
+    stageName: {
+      display: 'flex',
+      alignItems: 'center',
+    },
+    fieldEquality: {
+      display: 'flex',
+      alignItems: 'center',
+      '& span': {
+        color: 'white',
+        width: '20px',
+        height: '20px',
+        display: 'flex',
```

Comment:
what's the purpose of setting a span to display flex?

Reply:
I used it to horizontally center the span. But since span's content is text '=' instead of an icon, I simplified it now using text-align.